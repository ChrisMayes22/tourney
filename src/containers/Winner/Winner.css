.flexContainer__col{
    display: flex;
    margin: 5px auto;
    flex-direction: column;
    width: 95vw;
    height: 90vh;
    align-items: center;
    background-color: #a63939;
    background-color: #a63939;
    color: rgb(253, 211, 250);
    font-size: 1.3rem;
    border-radius: 10px;
}

h3{
    margin: 1.2rem 0 0.8rem 0;
}

h1, h5{
    margin: 0.3rem;
}

.grid {
    margin: 5px auto;
    display: grid;
    grid-template-rows: 15vh;
    grid-template-columns: 19vw 19vw 19vw 19vw 19vw;
    grid-template-areas: "header header header header header";
    grid-auto-rows: 30vh;
    grid-auto-columns: minmax(auto, 19vw);
    background-color: rgb(253, 211, 250);
    justify-items: center;
    align-items: center;
}

.mainImg{
    max-height: 70vh;
    max-width: 72vw;
}

.otherLosersHeader{
    grid-area: header;
    font-size: 2rem;
    color: #a63939;
}

.eliminationHeader{
    margin: 2rem;
}