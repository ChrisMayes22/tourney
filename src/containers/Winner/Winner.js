import React, { Component } from 'react';
import classes from './Winner.css';
import { connect } from 'react-redux';
import CharacterArray from '../../components/CharacterArray/CharacterArray';
import * as loserSettings from '../../components/CharacterArray/LoserSettings';

export class Winner extends Component{

    render(){
        return(
            <React.Fragment>
                <main className={classes.flexContainer__col}>
                    <h1 className={classes.eliminationHeader}>WINNER!!</h1>
                    <CharacterArray characters={this.props.characters} loserSettings={null} imageClass={classes.mainImg}/>
                </main>
                <section className={classes.grid}>
                    <h5 className={classes.otherLosersHeader}>~ The Losers ~</h5>
                    <CharacterArray characters={this.props.characters} loserSettings={loserSettings.DISPLAY_ALL_LOSERS}/>
                </section>
            </React.Fragment>
        );
    }
}

let mapStateToProps = state => {
    return {
        characters: state.characters
    }
};

export default connect(mapStateToProps, null)(Winner);