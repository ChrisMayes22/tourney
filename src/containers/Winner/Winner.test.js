import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { shallow } from 'enzyme';

import { Winner } from './Winner';

let props;
let wrapper;

describe('When unconnected component first renders', function(){
  beforeEach(function(){
    props = {characters: [{
      imageUrl: `foo$Image`,
      roundRatings: {
          rowOne: {rowId: `foo$id1-row-one`, points: 5},
          rowTwo: {rowId: `foo$id1-row-two`, points: 5},
          rowThree: {rowId: `foo$id1-row-three`, points: 5},
          rowFour: {rowId: `foo$id1-row-four`, points: 5}
      },
      points: 20,
      hadTurn: false,
      isEliminated: {check: false, whenEliminated: null},
      id: 'foo$id1'
    },
    {
      imageUrl: `foo$Image`,
      roundRatings: {
          rowOne: {rowId: `foo$id2-row-one`, points: 1},
          rowTwo: {rowId: `foo$id2-row-two`, points: 1},
          rowThree: {rowId: `foo$id2-row-three`, points: 1},
          rowFour: {rowId: `foo$id2-row-four`, points: 1}
      },
      points: 4,
      hadTurn: false,
      isEliminated: {check: true, whenEliminated: 0},
      id: 'foo$id2'
    },
    {
      imageUrl: `foo$Image`,
      roundRatings: {
          rowOne: {rowId: `foo$id3-row-one`, points: 2},
          rowTwo: {rowId: `foo$id3-row-two`, points: 2},
          rowThree: {rowId: `foo$id3-row-three`, points: 2},
          rowFour: {rowId: `foo$id3-row-four`, points: 2}
      },
      points: 8,
      hadTurn: false,
      isEliminated: {check: true, whenEliminated: 1},
      id: 'foo$id3'
    }],
    players: 3}
    
    wrapper = shallow(<Winner {...props}/>)
  })
    

    test('Winner renders as expected', () => {
      expect(wrapper).toMatchSnapshot();
    })
})