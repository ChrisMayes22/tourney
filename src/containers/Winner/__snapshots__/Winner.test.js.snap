// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When unconnected component first renders Winner renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Winner
    characters={
      Array [
        Object {
          "hadTurn": false,
          "id": "foo$id1",
          "imageUrl": "foo$Image",
          "isEliminated": Object {
            "check": false,
            "whenEliminated": null,
          },
          "points": 20,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 5,
              "rowId": "foo$id1-row-four",
            },
            "rowOne": Object {
              "points": 5,
              "rowId": "foo$id1-row-one",
            },
            "rowThree": Object {
              "points": 5,
              "rowId": "foo$id1-row-three",
            },
            "rowTwo": Object {
              "points": 5,
              "rowId": "foo$id1-row-two",
            },
          },
        },
        Object {
          "hadTurn": false,
          "id": "foo$id2",
          "imageUrl": "foo$Image",
          "isEliminated": Object {
            "check": true,
            "whenEliminated": 0,
          },
          "points": 4,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 1,
              "rowId": "foo$id2-row-four",
            },
            "rowOne": Object {
              "points": 1,
              "rowId": "foo$id2-row-one",
            },
            "rowThree": Object {
              "points": 1,
              "rowId": "foo$id2-row-three",
            },
            "rowTwo": Object {
              "points": 1,
              "rowId": "foo$id2-row-two",
            },
          },
        },
        Object {
          "hadTurn": false,
          "id": "foo$id3",
          "imageUrl": "foo$Image",
          "isEliminated": Object {
            "check": true,
            "whenEliminated": 1,
          },
          "points": 8,
          "roundRatings": Object {
            "rowFour": Object {
              "points": 2,
              "rowId": "foo$id3-row-four",
            },
            "rowOne": Object {
              "points": 2,
              "rowId": "foo$id3-row-one",
            },
            "rowThree": Object {
              "points": 2,
              "rowId": "foo$id3-row-three",
            },
            "rowTwo": Object {
              "points": 2,
              "rowId": "foo$id3-row-two",
            },
          },
        },
      ]
    }
    players={3}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <main>
          <h1>
            WINNER!!
          </h1>
          <div>
            <img
              alt="the winner!"
              src="foo$Image"
            />
          </div>
        </main>,
        <section>
          <h5>
            ~ The Losers ~
          </h5>
          <img
            alt="a losing competitor"
            src="foo$Image"
          />
          <img
            alt="a losing competitor"
            src="foo$Image"
          />
        </section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              WINNER!!
            </h1>,
            <div>
              <img
                alt="the winner!"
                src="foo$Image"
              />
            </div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "WINNER!!",
              "className": undefined,
            },
            "ref": null,
            "rendered": "WINNER!!",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="the winner!"
                  src="foo$Image"
                />,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "foo$id1",
                "nodeType": "host",
                "props": Object {
                  "alt": "the winner!",
                  "className": undefined,
                  "src": "foo$Image",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5>
              ~ The Losers ~
            </h5>,
            Array [
              <img
                alt="a losing competitor"
                src="foo$Image"
              />,
              <img
                alt="a losing competitor"
                src="foo$Image"
              />,
            ],
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "~ The Losers ~",
              "className": undefined,
            },
            "ref": null,
            "rendered": "~ The Losers ~",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": "foo$id2",
            "nodeType": "host",
            "props": Object {
              "alt": "a losing competitor",
              "className": undefined,
              "src": "foo$Image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "foo$id3",
            "nodeType": "host",
            "props": Object {
              "alt": "a losing competitor",
              "className": undefined,
              "src": "foo$Image",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "section",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <main>
            <h1>
              WINNER!!
            </h1>
            <div>
              <img
                alt="the winner!"
                src="foo$Image"
              />
            </div>
          </main>,
          <section>
            <h5>
              ~ The Losers ~
            </h5>
            <img
              alt="a losing competitor"
              src="foo$Image"
            />
            <img
              alt="a losing competitor"
              src="foo$Image"
            />
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                WINNER!!
              </h1>,
              <div>
                <img
                  alt="the winner!"
                  src="foo$Image"
                />
              </div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "WINNER!!",
                "className": undefined,
              },
              "ref": null,
              "rendered": "WINNER!!",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="the winner!"
                    src="foo$Image"
                  />,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "foo$id1",
                  "nodeType": "host",
                  "props": Object {
                    "alt": "the winner!",
                    "className": undefined,
                    "src": "foo$Image",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5>
                ~ The Losers ~
              </h5>,
              Array [
                <img
                  alt="a losing competitor"
                  src="foo$Image"
                />,
                <img
                  alt="a losing competitor"
                  src="foo$Image"
                />,
              ],
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "~ The Losers ~",
                "className": undefined,
              },
              "ref": null,
              "rendered": "~ The Losers ~",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": "foo$id2",
              "nodeType": "host",
              "props": Object {
                "alt": "a losing competitor",
                "className": undefined,
                "src": "foo$Image",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "foo$id3",
              "nodeType": "host",
              "props": Object {
                "alt": "a losing competitor",
                "className": undefined,
                "src": "foo$Image",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "section",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
